File: app/src/main/java/vergecurrency/vergewallet/viewmodel/ChartsViewModel.java
package vergecurrency.vergewallet.viewmodel;


import com.github.mikephil.charting.data.Entry;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.Set;

import androidx.lifecycle.ViewModel;
import vergecurrency.vergewallet.service.model.GraphInfo;
import vergecurrency.vergewallet.service.repository.StatsService;

public class ChartsViewModel extends ViewModel {

	public ChartsViewModel() {

	}


	public List<Entry> getVolumeData(int filter) {
		GraphInfo gi = StatsService.readPriceStatistics(filter);
		return createEntryList(gi.getVolume_usd().entrySet());
	}

	public List<Entry> getPriceData(int filter) {
		GraphInfo gi = StatsService.readPriceStatistics(filter);
		return createEntryList(gi.getPrice_usd().entrySet());
	}




	private List<Entry> createEntryList(Set<Map.Entry<String, String>> set) {
		List<Entry> entries = new ArrayList<>();
		float counter = 0;
		for (Map.Entry<String, String> e : set) {
			entries.add(new Entry(counter, Float.parseFloat(e.getValue())));
			counter++;
		}

		return entries;
	}
}

File: app/src/main/kotlin/vergecurrency/vergewallet/viewmodel/ChartsViewModel.kt
package vergecurrency.vergewallet.viewmodel


import androidx.lifecycle.ViewModel
import com.github.mikephil.charting.data.Entry
import vergecurrency.vergewallet.service.model.ChartInfo
import vergecurrency.vergewallet.service.repository.StatsService
import java.util.*

class ChartsViewModel : ViewModel() {

    var gi: ChartInfo? = null

    fun retrieveData(filter: Int) {
        gi = StatsService.readPriceStatistics(filter)
    }

    fun getVolumeData(): List<Entry> {
        return createEntryList(gi!!.volume_usd!!.entries)
    }

    fun getPriceData(): List<Entry> {
        return createEntryList(gi!!.price_usd!!.entries)
    }


    private fun createEntryList(set: Set<Map.Entry<String, String>>): List<Entry> {
        val entries = ArrayList<Entry>()
        var counter = 0f
        for (e in set) {
            entries.add(Entry(counter, java.lang.Float.parseFloat(e.value)))
            counter++
        }

        return entries
    }
}
